test/spoof/spoof.go

```diff
@@ -66,7 +66,7 @@ type ResponseChecker func(resp *Response) (done bool, err error)
 
 // ErrorRetryChecker is used to determine if an error should be retried or not.
 // If an error should be retried, it should return true and the wrapped error to explain why to retry.
-type ErrorRetryChecker func(e error) (retry bool, err error)
+type ErrorRetryChecker func(resp *Response, e error) (retry bool, err error)
```

Comment:
Maybe introduce a new type `ResponseRetryChecker` ? And have Do and Poll accept both `ErrorRetryChecker` and `ResponseRetryChecker` ?

Reply:
We could, though tbh. I dunno how valuable that'd be as we'd have to do type casting funkiness and whatnot to tell the two apart. Do you feel like that's going to be clearer?